router-tests/telemetry/telemetry_test.go

```diff
@@ -9454,3 +9454,1889 @@ func TestFlakyTelemetry(t *testing.T) {
 	})
 
 }
+
+func TestExcludeAttributesWithCustomExporter(t *testing.T) {
+	const (
+		UseCloudExporter                           = "use_cloud_exporter"
+		UseCustomExporterOnly                      = "use_custom_exporter_only"
+		UseCustomExporterWithRouterConfigAttribute = "use_custom_exporter_with_router_config_attribute"
+	)
+
+	appendRouterConfigAndOperationInfoConditionally := func(before []attribute.KeyValue, useCustomExporter string, routerConfigVersion string, addOperationInfo bool) []attribute.KeyValue {
+		var mid []attribute.KeyValue
+		if useCustomExporter == UseCloudExporter || useCustomExporter == UseCustomExporterWithRouterConfigAttribute {
+			mid = append(mid, otel.WgRouterConfigVersion.String(routerConfigVersion))
+		}
+		if useCustomExporter == UseCloudExporter && addOperationInfo {
+			mid = append(mid,
+				otel.WgOperationHash.String("1163600561566987607"),
+				otel.WgOperationName.String(""),
+			)
+		}
+		return append(before, mid...)
+	}
+
+	t.Run("Verify metrics when there is a router config version metric attribute", func(t *testing.T) {
```

Comment:
Can we simplify this? These are few K lines of code just to validate if we have an attribute set. It's super hard for me to understand what these tests really tests.

Reply:
I removed all attributes, since base otels are set to every attribute. We validate one.